src/satipc.c

```diff
@@ -1110,6 +1119,7 @@ int http_request(adapter *ad, char *url, char *method, int force) {
     if (sip->stream_id != -1)
         sprintf(sid, "stream=%d", sip->stream_id);
     else if (!strcmp(method, "TEARDOWN")) {
+        sip->uncommitted_sid = 0;
```

Comment:
Here the value is set to 0, everywhere else is -1? Is there a need for 2 values or 1 value (-1) is enough?

Reply:
Hi @catalinii ,

The use of the member `uncommitted_sid` is this:
- `-1` : init value, no TEARDOWN send, so regular behaviour.
- `0`  : TEARDOWN command delayed, pending to read the "stream_id".
- `>0` : Value of the "stream_id" active because a TEARDOWN command is not sended.

So, the short response to your question is: NO.

Long answer: The TEARDOWN command can be delayed, not because the module wants to delay it, but because the RESPONSE of the previous command has not yet been received. Then you can't close the session as it's possible you don't know the "stream_id". So the code waits until the pending command is answered and then it copy the "stream_id" in the `uncommitted_sid` variable. Then the execution continues reusing this value in the next command (instead of sending a new SETUP it send a PLAY with the same "stream_id").

The code is then correct. I'm using it for several days and the behavious has improved a lot. So, please commit it.
Regards.

Reply:
@lars18th I do see the issue you are mentioning happening and I agree it is a problem. 

And thinking about it is that I am using an async way of communicating with the server by caching future commands instead of waiting for the answer.
Eg. Send tune, wait for answer then respond to the satip client.

But the issue with this patch is more that feels like a band aid solution instead of fixing the scenario: PLAY after a cached TEARDOWN. 
I also agree that handling of stream_id and session is unfortunate. It should be cleared only when TEARDOWN is sent.

I would prefer fixing this scenario:
- fix habdling of session, stream_id with no new variables
- if a teardown is cached and a play comes, delete the cached teardown.


Otherwise by accepting the patch It would be very hard to fix if a new issue ie discovered in the previous code.

Reply:
Hi @catalinii ,

Sorry, but I feel I can't solve the issue using what you propose. Let me explain why:

- The current code is only detecting an edge case: when the module is instructed to send a TEADOWN message and it can't be sended as the stream_id is unkown, because the precedent command is pending and the response has not yet been received.
- So the code only targets this specific case with minimal changes and without adding innecessary variables.

Then my questions are:
- How I can handle this case without new variables? I need a flag to know when it appears, and I only added one int var (`uncommitted_sid`) and a cache of the session id string (`uncommitted_session`).
- I don't know how to "delete the cached TEARDOWN command". In fact, as the command is not sended we don't need to cache anything. When the next command is processed we only need to know if we start from a "stop state" or from "resume state" (aka delayed teardown).

Regards.

Reply:
This is an example of deleting cached OPTIONS:
https://github.com/catalinii/minisatip/blob/master/src/satipc.c#L276